packages/expo-localization/android/src/main/java/expo/modules/localization/LocalizationModule.kt

```diff
@@ -73,6 +87,72 @@ class LocalizationModule : Module() {
       getCountryCode(locale)
     }
   }
+
+  private fun getMeasurementSystem(locale: Locale): String? {
+    return if (VERSION.SDK_INT >= VERSION_CODES.P) {
+      mapOf<LocaleData.MeasurementSystem, String>(
+        LocaleData.MeasurementSystem.SI to "metric",
+        LocaleData.MeasurementSystem.UK to "uk",
+        LocaleData.MeasurementSystem.US to "us"
+      )[LocaleData.getMeasurementSystem(ULocale.forLocale(locale))]
+    } else {
+      if (getRegionCode(locale).equals("uk")) return "uk"
+      if (USES_IMPERIAL.contains(getRegionCode(locale))) return "us" else return "metric"
+    }
```

Comment:
```suggestion
    return if (VERSION.SDK_INT >= VERSION_CODES.P) {
      when (LocaleData.getMeasurementSystem(ULocale.forLocale(locale))) {
        LocaleData.MeasurementSystem.SI -> "metric"
        LocaleData.MeasurementSystem.UK -> "uk",
        LocaleData.MeasurementSystem.US -> "us"
      }
    } else {
      if (getRegionCode(locale).equals("uk")) "uk"
      else if (USES_IMPERIAL.contains(getRegionCode(locale))) "us" 
      else "metric"      
    }
```

By writing this function like this it always returns a non-nullable string.